src/Connections/GithubRepo.ts

```diff
@@ -1973,47 +1987,28 @@ export class GitHubRepoConnection
       org: this.org,
       repo: this.repo,
     });
-    // Do a sanity check that the event exists.
-    try {
-      await this.intent.underlyingClient.getRoomStateEvent(
-        this.roomId,
-        GitHubRepoConnection.CanonicalEventType,
-        this.stateKey,
-      );
-      await this.intent.underlyingClient.sendStateEvent(
-        this.roomId,
-        GitHubRepoConnection.CanonicalEventType,
-        this.stateKey,
-        { disabled: true },
-      );
-    } catch (ex) {
-      await this.intent.underlyingClient.getRoomStateEvent(
-        this.roomId,
-        GitHubRepoConnection.LegacyCanonicalEventType,
-        this.stateKey,
-      );
-      await this.intent.underlyingClient.sendStateEvent(
-        this.roomId,
-        GitHubRepoConnection.LegacyCanonicalEventType,
-        this.stateKey,
-        { disabled: true },
-      );
-    }
+    await removeConnectionState(
+      this.intent.underlyingClient,
+      this.roomId,
+      this.stateKey,
+      GitHubRepoConnection,
+    );
   }
 
-  private matchesLabelFilter(itemWithLabels: {
-    labels?: { name: string }[];
-  }): boolean {
-    const labels = itemWithLabels.labels?.map((l) => l.name) || [];
-    if (this.state.excludingLabels?.length) {
-      if (this.state.excludingLabels.find((l) => labels.includes(l))) {
-        return false;
-      }
-    }
-    if (this.state.includingLabels?.length) {
-      return !!this.state.includingLabels.find((l) => labels.includes(l));
-    }
-    return true;
```

Comment:
Any reason why this was moved up in the file?

Reply:
Errrm, not sure. I think to move it nearer where it was being used. Can revert if it's worse